File: back/Domain/Entities/System/BusinessesCompanies/CompanyProfile.cs
using Domain.Entities.Shared;
using Domain.Entities.System.Customers;

namespace Domain.Entities.System.BusinessesCompanies;

public class CompanyProfile : RootBase
{
    
    public required string CompanyAuthId { get; set; }
    public  int BusinessProfileId { get; set; }
    public required string CNPJ { get; set; }
    public BusinessProfile? BusinessProfile { get; set; }
    public Address? Address { get; set; }
    public Contact? Contact { get; set; } = new();
    public ICollection<CustomerCompany> CustomersCompanies { get; set; } = new List<CustomerCompany>();

}

File: front/src/components/company/dtos/company-profile.ts
import { AddressDto } from "shared/components/address/dtos/address-dto";
import { ContactDto } from "shared/components/contact/dtos/contact-dto";

export class CompanyProfile {
  companyAuthId!: number;
  businessProfileId!: number;
  businessProfile!: number;
  address!: AddressDto;
  contact!: ContactDto;
  deleted!: Date;
  Registered!: Date;
}
